pyccel/codegen/printing/cwrappercode.py

```diff
@@ -243,10 +296,11 @@ def _body_optional_variable(self, tmp_variable, variable, collect_var, check_typ
         body = [(PyccelEq(VariableAddress(collect_var), VariableAddress(Py_None)),
                 [Assign(VariableAddress(variable), Nil())])]
         if check_type : # Type check
-            check = PyccelNot(PythonType_Check(variable, collect_var))
+            check = PyccelNot(PyccelAssociativeParenthesis(PyccelOr(NumpyType_Check(variable, collect_var)
+                    , PythonType_Check(variable, collect_var))))
```

Comment:
You shouldn't need to add parentheses yourself now that PR #559 has been merged

Reply:
ok I will change it